File: EDR_NDR/frontend/src/types/virustotal.ts
export interface VTFileResult {
  hash: string
  detection_ratio: string
  malicious_count: number
  total_engines: number
  malicious_vendors: string[]
  file_type: string | null
  meaningful_name: string | null
  first_seen: string | null
  cached: boolean
  checked_at: string
}

export interface VTIpResult {
  ip: string
  detection_ratio: string
  malicious_count: number
  total_engines: number
  malicious_vendors: string[]
  reputation: number | null
  country: string | null
  as_owner: string | null
  cached: boolean
  checked_at: string
}

export interface VTDomainResult {
  domain: string
  detection_ratio: string
  malicious_count: number
  total_engines: number
  malicious_vendors: string[]
  reputation: number | null
  categories: string[]
  creation_date: string | null
  cached: boolean
  checked_at: string
}

export interface VTUrlResult {
  url: string
  status: 'queued' | 'completed'
  detection_ratio: string | null
  malicious_count: number | null
  total_engines: number | null
  malicious_vendors: string[]
  cached: boolean
  checked_at: string
}

export interface VTQuota {
  hourly_used: number | null
  hourly_allowed: number | null
  daily_used: number | null
  daily_allowed: number | null
  monthly_used: number | null
  monthly_allowed: number | null
}

File: EDR_NDR/backend/app/schemas/virustotal.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel


class VTFileResult(BaseModel):
    hash: str
    detection_ratio: str
    malicious_count: int
    total_engines: int
    malicious_vendors: list[str]
    file_type: str | None
    meaningful_name: str | None
    first_seen: datetime | None
    cached: bool
    checked_at: datetime
    raw: dict[str, Any]


class VTIpResult(BaseModel):
    ip: str
    detection_ratio: str
    malicious_count: int
    total_engines: int
    malicious_vendors: list[str]
    reputation: int | None
    country: str | None
    as_owner: str | None
    cached: bool
    checked_at: datetime
    raw: dict[str, Any]


class VTDomainResult(BaseModel):
    domain: str
    detection_ratio: str
    malicious_count: int
    total_engines: int
    malicious_vendors: list[str]
    reputation: int | None
    categories: list[str]
    creation_date: datetime | None
    cached: bool
    checked_at: datetime
    raw: dict[str, Any]


class VTUrlSubmitRequest(BaseModel):
    url: str


class VTUrlResult(BaseModel):
    url: str
    status: str  # "queued" | "completed"
    detection_ratio: str | None
    malicious_count: int | None
    total_engines: int | None
    malicious_vendors: list[str]
    cached: bool
    checked_at: datetime
    raw: dict[str, Any]


class VTQuota(BaseModel):
    hourly_used: int | None
    hourly_allowed: int | None
    daily_used: int | None
    daily_allowed: int | None
    monthly_used: int | None
    monthly_allowed: int | None
